packages/neptune-extension/src/common/hooks/useSelectInputValue.ts

```diff
@@ -9,15 +9,14 @@ export interface SelectInputProps {
 }
 
 export interface SelectMetaProps {
-  valid?: boolean
-  loading?: boolean
+  status: 'pending' | 'success' | 'error'
```

Comment:
This code is meant to be two flags, one representing the promise (loading), and the other representing the validity (valid). You rewrote it to one promise flag, which is something different logically. It is difficult to see the reason behind this change, because the old interface supports the usecase you were going to implement. This PR says "Style Select", but you rewrote a lot of unrelated code.